src/asl.ts

```diff
@@ -1293,11 +1301,11 @@ export namespace ASL {
       return expr.value ?? null;
     } else if (expr.kind === "ReferenceExpr") {
       const ref = expr.ref();
-      if (ref.kind === "Function") {
+      if (isFunction(ref)) {
```

Comment:
So this is in part because ref is no longer just for services. It can now return anything, so the type essentially `unknown`.

```ts
export type CanReference =
  | AnyTable
  | AnyLambda
  | AnyStepFunction
  | typeof $AWS
  | unknown;
```

The guards work for `unknown`, but `.kind` does not.

```ts
const sfn = new StepFunction();
new EventBridge.when(event => sfn.stepFunctionArn);
```

=>

```ts
propAccess(ref(() => sfn), "stepFunctionArn")
```